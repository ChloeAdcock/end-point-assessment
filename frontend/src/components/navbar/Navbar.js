import React, { useEffect } from "react";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Button from "@material-ui/core/Button";
import { Link } from "react-router-dom";
import { useSelector, useDispatch } from "react-redux";
import { currentUser, logout } from "../../redux/actions/accounts/accounts";

function Navbar() {
  const dispatch = useDispatch();
  const user = useSelector((state) => state.accounts.currentUser);

  useEffect(() => {
    dispatch(currentUser());
  }, []);

  const handleClick = () => {
    dispatch(logout());
  };

  return (
    <div>
      <AppBar position="fixed">
        <Toolbar>
          <Typography variant="h6" className={classes.title}>
            Community Events
          </Typography>
          {user ? (
            <div>
              <Button component={Link} to="/home" color="inherit">
                Community Events
              </Button>
              <Button component={Link} to="/newevent" color="inherit">
                New Event
              </Button>
              <Button
                onClick={handleClick}
                component={Link}
                to="/login"
                color="inherit"
              >
                Logout
              </Button>
            </div>
          ) : (
            <div>
              <Button component={Link} to="/login" color="inherit">
                Login
              </Button>
              <Button component={Link} to="/register" color="inherit">
                Register
              </Button>
            </div>
          )}
        </Toolbar>
      </AppBar>
      {/* Empty toolbar to prevent content being rendered under navbar */}
      <Toolbar />
    </div>
  );
}

export default Navbar;
