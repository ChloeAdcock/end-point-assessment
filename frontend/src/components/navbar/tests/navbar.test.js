import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Navbar from "../Navbar";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe("The navbar component", () => {
  it("should render the unauthenticated navbar when a user is not authenticated", () => {
    const store = mockStore({
      accounts: {
        currentUser: null,
      },
    });
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <Navbar />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render the authenticated navbar when a user is authenticated", () => {
    const store = mockStore({
      accounts: {
        currentUser: "test",
      },
    });
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <Navbar />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
