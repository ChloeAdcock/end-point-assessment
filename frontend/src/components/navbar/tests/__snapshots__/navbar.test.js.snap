// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The navbar component should render the authenticated navbar when a user is authenticated 1`] = `
<div>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appbar-2 mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        Community Events
      </h6>
      <div
        className="makeStyles-navbar-1"
      >
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          href="/home"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            Home
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          href="/newevent"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            New Event
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          href="/"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            Logout
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </header>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  />
</div>
`;

exports[`The navbar component should render the unauthenticated navbar when a user is not authenticated 1`] = `
<div>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appbar-2 mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        Community Events
      </h6>
      <div
        className="makeStyles-navbar-1"
      >
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          href="/"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            Login
          </span>
        </a>
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          href="/register"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            Register
          </span>
        </a>
      </div>
    </div>
  </header>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  />
</div>
`;
